modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -838,6 +856,7 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
     val closing = if (recursive) ")" else ""
     lines(
       deprecationAnnotation(hints),
+      documentationAnnotation(hints),
       obj(name, line"$Newtype_[$tpe]")(
         renderId(shapeId),
         renderHintsVal(hints),
```

Comment:
(not related to this line in particular)

I think we're missing an example (and possibly an impl) for documenting structure members...

I guess this could look better if we do `@param` inside the structure's Scaladoc, as the lines would get quite lengthy if we just put it on the case class fields themselves. Maybe let's add this in a future PR.

In the meantime, I'd like to see an example of a documented `alt` case in a union as well. Ideally, every usage of `documentationAnnotation(hints)` would be checked with an example.

Reply:
So should I make a `DocumentationService.smithy` file that has all these variants of docs on members/shapes/unions?

Reply:
That works, but I meant that we need these cases to be test covered - not necessarily as part of a new spec, reusing existing ones is fine